### ISSUE TITLE
Mark seald trait @DoNotInherent

### ISSUE THREAD

**Issue by He-Pin** [apache/pekko#1270]
There are classes is not marked with `@DoNotInherent`, then Java user may extends that classes. We should mark it with `@DoNotInherent`. 

And there are some classes should be annotated with `@InternalApi`.

**Comment by mdedetrich** [comment]
Is `@DoNotInherent` necessary if the class is final, that is already implied?

Typically `@ DoNotInherent` is used for classes which can be extended but only for private cases

**Comment by He-Pin** [comment]
no, I mean we should mark on sealed trait and sealed abstract class

**Comment by mdedetrich** [comment]
> no, I mean we should mark on sealed trait and sealed abstract class

That makes perfect sense, :+1 from me
